cmake/support_catalyst.cmake

```diff
@@ -0,0 +1,125 @@
+###############################################################################################
+# This file provides macros to process Catalyst.
+###############################################################################################
+
+# Include this only once
+include_guard()
+
+macro(FindCatalyst target_name)
+    if(LIGHTNING_CATALYST_SRC_PATH)
+        if(NOT IS_ABSOLUTE ${LIGHTNING_CATALYST_SRC_PATH})
+            message(FATAL_ERROR " LIGHTNING_CATALYST_SRC_PATH=${LIGHTNING_CATALYST_SRC_PATH} must be set to an absolute path")
+        endif()
+        if(CATALYST_GIT_TAG)
+            message(WARN " Setting `LIGHTNING_CATALYST_SRC_PATH=${LIGHTNING_CATALYST_SRC_PATH}` overrides `CATALYST_GIT_TAG=${CATALYST_GIT_TAG}`")
+        endif()
+
+        # Acquire local git hash and use for CATALYST_GIT_TAG
+        execute_process(COMMAND git rev-parse --short HEAD
+            WORKING_DIRECTORY ${LIGHTNING_CATALYST_SRC_PATH}
+            OUTPUT_VARIABLE CATALYST_GIT_TAG
+        )
+        message(INFO " Building against local Catalyst - path: ${LIGHTNING_CATALYST_SRC_PATH} - GIT TAG: ${CATALYST_GIT_TAG}")
+
+        target_include_directories(${target_name} PUBLIC ${LIGHTNING_CATALYST_SRC_PATH}/runtime/lib/backend/common)
+        target_include_directories(${target_name} PUBLIC ${LIGHTNING_CATALYST_SRC_PATH}/runtime/lib/capi)
+        target_include_directories(${target_name} PUBLIC ${LIGHTNING_CATALYST_SRC_PATH}/runtime/include)
+        target_include_directories(${target_name} PUBLIC ${LIGHTNING_CATALYST_SRC_PATH}/runtime/utils)
+        target_include_directories(${target_name} PUBLIC ${LIGHTNING_CATALYST_SRC_PATH}/runtime/tests)
+
+    else()
+        if(NOT CATALYST_GIT_TAG)
+            #set(CATALYST_GIT_TAG "main" CACHE STRING "GIT_TAG value to build Catalyst")
+            # Build against catalyst PR https://github.com/PennyLaneAI/catalyst/pull/1273
+            # to apply lightning CI clang tidy to catalyst files that won't be moved
+            set(CATALYST_GIT_TAG "clang_tidy_runtime_capi")
```

Comment:
revert back to main when clang-tidy is done, and PR in catalyst is merged to main